src/Orchard.Cms.Web/Modules/Orchard.OpenId/Services/OpenIdService.cs

```diff
@@ -55,36 +58,36 @@ public bool IsValidOpenIdSettings(OpenIdSettings settings, ModelStateDictionary
             {
                 if (settings.CertificateStoreName == null)
                 {
-                    modelState.AddModelError("CertificateStoreName", "A Certificate Store Name is required.");
+                    modelState.AddModelError("CertificateStoreName", T["A Certificate Store Name is required."]);
                 }
                 if (settings.CertificateStoreLocation == null)
                 {
-                    modelState.AddModelError("CertificateStoreLocation", "A Certificate Store Location is required.");
+                    modelState.AddModelError("CertificateStoreLocation", T["A Certificate Store Location is required."]);
                 }
                 if (string.IsNullOrWhiteSpace(settings.CertificateThumbPrint))
                 {
-                    modelState.AddModelError("CertificateThumbPrint", "A certificate is required when testing mode is disabled.");
+                    modelState.AddModelError("CertificateThumbPrint", T["A certificate is required when testing mode is disabled."]);
                 }
             }
             return modelState.IsValid;
         }
 
-        private static void ValidateUrisSchema(IEnumerable<string> uriStrings, bool onlyAllowHttps, ModelStateDictionary modelState, string modelStateKey)
+        private void ValidateUrisSchema(IEnumerable<string> uriStrings, bool onlyAllowHttps, ModelStateDictionary modelState, string modelStateKey)
         {
             if (uriStrings == null)
             {
-                modelState.AddModelError(modelStateKey, "Invalid url.");
+                modelState.AddModelError(modelStateKey, T["Invalid url."]);
                 return;
             }
-            foreach (var uriString in uriStrings.Select(a=> a??"".Trim()))
+            foreach (var uriString in uriStrings.Select(a => a ?? "".Trim()))
             {
                 Uri uri;
-                if (!Uri.TryCreate(uriString, UriKind.Absolute, out uri) || ((onlyAllowHttps || uri.Scheme!="http") && uri.Scheme!="https"))
+                if (!Uri.TryCreate(uriString, UriKind.Absolute, out uri) || ((onlyAllowHttps || uri.Scheme != "http") && uri.Scheme != "https"))
                 {
                     var message = "Invalid url.";
                     if (onlyAllowHttps)
                         message += " Non https urls are only allowed in testing mode.";
-                    modelState.AddModelError(modelStateKey, message);
+                    modelState.AddModelError(modelStateKey, T[message]);
```

Comment:
You can't do that, don't translate a dynamically created string. Or it will be hard to create the translation. 
Do this instead:
```
var message = onlyAllowHttps 
? T["Invalid url."]
: T["nvalid url. Non https urls are only allowed in testing mode."];
```